pkg/aws/policies.go

```diff
@@ -162,8 +190,37 @@ func (c *awsClient) createRole(name string, policy string, tagList map[string]st
 		}
 		return "", err
 	}
-	role := output.Role
-	return aws.StringValue(role.Arn), nil
+	return aws.StringValue(output.Role.Arn), nil
+}
+
+func (c *awsClient) isRoleCompatible(name string, version string) (bool, error) {
+	output, err := c.iamClient.ListRoleTags(&iam.ListRoleTagsInput{
+		RoleName: aws.String(name),
+	})
+	if err != nil {
+		return false, err
+	}
+	if len(output.Tags) == 0 {
+		return false, err
+	}
+	for _, tag := range output.Tags {
+		if aws.StringValue(tag.Key) == tags.OpenShiftVersion {
+			if version == aws.StringValue(tag.Value) {
+				return true, nil
+			}
+			wantedVersion, err := semver.NewVersion(version)
+			if err != nil {
+				return false, err
+			}
+			currentVersion, err := semver.NewVersion(aws.StringValue(tag.Value))
+			if err != nil {
+				return false, err
+			}
+			return currentVersion.GreaterThanOrEqual(wantedVersion), nil
+		}
+	}
+
+	return false, err
```

Comment:
This block is repeated in `isPolicyCompatible` might be worth breaking out into a seperate function.

Reply:
Ah, good point. I'll do that :+1:

Reply:
Done